src/crypto/CHIPCryptoPALOpenSSL.cpp

```diff
@@ -214,6 +226,18 @@ CHIP_ERROR AES_CCM_decrypt(const uint8_t * ciphertext, size_t ciphertext_length,
     int result               = 1;
     const EVP_CIPHER * type  = nullptr;
 
+    const uint8_t * empty_ciphertext = Uint8::from_const_char("");
+
+    if (ciphertext_length != 0)
+    {
+        VerifyOrExit(ciphertext != nullptr, error = CHIP_ERROR_INVALID_ARGUMENT);
+        VerifyOrExit(plaintext != nullptr, error = CHIP_ERROR_INVALID_ARGUMENT);
```

Comment:
This should always be true as well.

Reply:
Resolving after slack conversation with @tcarmelveilleux . The nullptr for plain text works if cipher text length is 0. The unit tests are covering this condition.